airflow/providers/google/cloud/hooks/bigquery.py

```diff
@@ -69,8 +69,8 @@ class BigQueryHook(GoogleBaseHook, DbApiHook):
 
     conn_name_attr = 'gcp_conn_id'
     default_conn_name = 'google_cloud_default'
-    conn_type = 'google_cloud_platform'
-    hook_name = 'Google Cloud'
+    conn_type = 'bigquery'
+    hook_name = 'BigQuery'
```

Comment:
@turbaszek @potiuk @mik-laj is this change ok?

Reply:
@xinbinhuang Do we need this change?

Reply:
I left a comment about this previously, but I think it got buried over time. Here I copy my original comment here.

This fixes the issue where conn.get_hook will never return the bigquery hook because GoogleBaseHook "hides" the BigQueryHook as they have the same conn_type.

> Yes, there is one place that may be considered fixing rather than breaking. For airflow/providers/google/cloud/hooks/bigquery.py, I changed the hook's conn_type and hook_name from google cloud platform/Google Cloud to bigquery/BigQuery. This fixes the issue where conn.get_hook will never return the bigquery hook because GoogleBaseHook "hides" the BigQueryHook as they have the same conn_type.

Reply:
Does it needs any action from a user side is why I thinking about it -- if yes, then we might need a note somewhere in Up:

https://github.com/apache/airflow/blob/master/airflow/providers/google/cloud/ADDITIONAL_INFO.md

Reply:
As similar to other Google cloud connection type:

![image](https://user-images.githubusercontent.com/27927454/109046868-12ecb800-768a-11eb-970c-6d52905825fa.png)
![image](https://user-images.githubusercontent.com/27927454/109046902-1bdd8980-768a-11eb-9cc2-155f26ef7898.png)

Maybe I should rename it to `gcpbigquery` & `Google Bigquery` instead?

Reply:
Renamed, this is the new connect_type list for google:
![image](https://user-images.githubusercontent.com/27927454/109047564-de2d3080-768a-11eb-95ee-6e614ae6c27a.png)

Reply:
> Does it needs any action from a user side is why I am thinking about it ? -- if yes, then we might need a note somewhere in Up: https://github.com/apache/airflow/blob/master/airflow/providers/google/cloud/ADDITIONAL_INFO.md

Hmm, I think we need a note for it. From sole user perspective, the conn_type for BigQuery changes from `Google Cloud` -> `Google BigQuery`.

Should I update the providers' version by 0.0.1 and add a note to the `ADDITIONAL_INFO.md` or other documentation?

Reply:
Yes probably add an Updating.md style note in `ADDITIONAL_INFO.md` with version ad header -- same as https://github.com/apache/airflow/blob/master/UPDATING.md

Reply:
or in https://github.com/apache/airflow/blob/master/airflow/providers/google/CHANGELOG.rst actually

Reply:
Just a comment on this one. Yeah. I wanted to avoid this change when we switched to 2.0 (not to add backwards incompatibilities), but this change indeed breaks the compatibility of Google Provider, so we should mention that in the CHANGELOG @kaxil mentioned. I think however, whether this is a good time to "merge" this change now - we are planning a bigger release of Google Provider (3.0.0) around March/April when we will drop "backports" altogether. We need to update dependencies and upgrade a few remaining APIs of python libraries of Google - and such a backwards-incompatible change could be added then. 

I think if we could make this change without this breaking compatibility now, that would be ideal.

Reply:
@potiuk I want to clarify a bit about the change here. The current Goold Providers in the master branch is broken because it doesn't return  `BigQueryHook` when we do `Connection.get_hook` (this is the original behavior). 

Before these 2 PRs #12466 #12558, `Connection.get_hook` was using a static `conn_type` -> `HookClass` mapping to get the underlying hook. However, after #12558, `airflow.providers.google.common.hooks.base_google.GoogleBaseHook` was added after `BigQueryHook` so `BigQueryHook` will be overridden during providers discovery. https://github.com/PolideaInternal/airflow/blob/802159767baf1768d92c6047c2fdb2094ee7a2a8/airflow/providers/google/provider.yaml#L714-L720

To solve this, I think we can either:
- Fix the bug by putting  `BigQueryHook`  after `GoogleBaseHook`. This will give back us the old behavior.
- Rename the `conn_type` as in this PR, which will involve some versioning consideration.

I think we can fix the bug in this PR, and create a new issue/PR with the new `conn_type`. I am not familiar with the overall releasing schedule and planning, so please let me know which one do you prefer? @potiuk @kaxil

Reply:
I have reverted the `BigQueryHook` conn_type change and put `BigQueryHook` after `GoogleBaseHook` for now.